Controllers/ParkedVehiclesController.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                Text = v.ToString(),
+                Value = ((int)v).ToString()
+            });
+            ViewBag.VehicleTypes = new SelectList(vehicleTypes, "Value", "Text");
+            return View();
+        }
+        [HttpPost]
+        public async Task<ActionResult> Parking(ParkedVehicle model)
+        {
+            if(ModelState.IsValid)
+            {
+                //Automatically set arrival time to current date and time
+                model.ArrivalTime = DateTime.Now;
+
+                //Implementation for parking vehicle
+                _context.ParkedVehicles.Add(model);
+                await _context.SaveChangesAsync();
+
+                TempData["Message"] = "Vehicle Parked successfully!";
+                return RedirectToAction("Overview");
+            }
+            var vehicleTypes = Enum.GetValues(typeof(VehicleType)).Cast<VehicleType>().Select(v => new SelectListItem
+            {
+                Text = v.ToString(),
+                Value = ((int)v).ToString()
+            });
+            ViewBag.VehicleTypes = new SelectList(vehicleTypes, "Value", "Text");
+            return View(model);
+        }
+
+        public ActionResult Retrieving()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> Retrieving(string registrationNumber)
+        {
+            var vehicle = await _context.ParkedVehicles.FirstOrDefaultAsync(v => v.RegistrationNumber == registrationNumber);
+            if(vehicle != null)
+            {
+                _context.ParkedVehicles.Remove(vehicle);
+                await _context.SaveChangesAsync();
+
+                TempData["Message"] = "vehicle retrived successfuly!";
+
+            }
+            else
+            {
+                TempData["Message"] = "vehicle not found!";
+            }
+            return RedirectToAction("Overview");
+        }
+
+        public ActionResult Edit(int id)
+        {
+            var vehicle = _context.ParkedVehicles.Find(id);
+            return View(vehicle);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> Edit(ParkedVehicle model)
+        {
+            var existingVehicle = await _context.ParkedVehicles.FindAsync(model.Id);
+            if (existingVehicle != null)
+            {
+                existingVehicle.VehicleType = model.VehicleType;
+                existingVehicle.Color = model.Color;
+                existingVehicle.Brand = model.Brand;
+                existingVehicle.Model = model.Model;
+                existingVehicle.NumberOfWheels = model.NumberOfWheels;
+                await _context.SaveChangesAsync();
+
+                TempData["Message"] = "Vehicle edited successfully!";
+                return RedirectToAction("Overview");
+            }
+            else
+            {
+                TempData["Message"] = "Vehicle not found!";
+                return View(model);
+            }
+        }
+
```

Comment:
Shouldn't we have the [HttpGet] inside here? For displaying the data without a change.